# HW4 Task 2 Fire Perimeter

**Link to github repo:** https://github.com/haylee360/eds220-hwk4 

### Fire perimeter data retrieval and selection

Explore the data and write a brief summary of the information you obtained from the preliminary information. Your summary should include the CRS of the data and whether this is projected or geographic.

In [1]:
# Load libraries
import os
import pandas as pd
import matplotlib.pyplot as plt
import geopandas as gpd

# Change display settings to see all column names
pd.set_option("display.max.columns", None)

# Import fire perimeter data
# perimeters = gpd.read_file(os.path.join('no_push_data',
#                                     'calfire_all.gdb')
#                                     )

perimeters = gpd.read_file(os.path.join('no_push_data',
                                    'California_Fire_Perimeters_(all)',
                                    'California_Fire_Perimeters_(all).shp')
                                    )


In [2]:
# View the first five rows of the perimeter data frame
perimeters.head()

OBJECTID   YEAR_ STATE AGENCY UNIT_ID  FIRE_NAME   INC_NUM  ALARM_DATE  \
0         1  2023.0    CA    CDF     SKU  WHITWORTH  00004808  2023-06-17   
1         2  2023.0    CA    LRA     BTU     KAISER  00010225  2023-06-02   
2         3  2023.0    CA    CDF     AEU    JACKSON  00017640  2023-07-01   
3         4  2023.0    CA    CDF     AEU     CARBON  00018821  2023-07-11   
4         5  2023.0    CA    CDF     AEU    LIBERTY  00018876  2023-07-11   

    CONT_DATE  CAUSE  C_METHOD  OBJECTIVE  GIS_ACRES COMMENTS COMPLEX_NA  \
0  2023-06-17    5.0       1.0        1.0   5.729125     None       None   
1  2023-06-02    5.0       1.0        1.0  13.602380     None       None   
2  2023-07-02    2.0       1.0        1.0  27.814460     None       None   
3  2023-07-11    9.0       1.0        1.0  58.760230     None       None   
4  2023-07-12   14.0       1.0        1.0  70.979000     None       None   

                                  IRWINID FIRE_NUM COMPLEX_ID  DECADES  \
0  {7985848C-0AC2-4BA4-8F0E-29F778652E61}     None       None   2020.0   
1  {43EBCC88-B3AC-48EB-8EF5-417FE0939CCF}     None       None   2020.0   
2  {B64E1355-BF1D-441A-95D0-BC1FBB93483B}     None       None   2020.0   
3  {CB41DB0A-E4B1-489D-A4EA-738F2CD6DB3B}     None       None   2020.0   
4  {F83F70A4-07A7-40B8-BD51-10CCC1C30D63}     None       None   2020.0   

      Shape__Are   Shape__Len  \
0   41407.839844  1247.166034   
1   93455.878906  1285.514550   
2  183028.500000  2697.587429   
3  388221.953125  2548.738779   
4  466456.007812  5106.936526   

                                            geometry  
0  POLYGON ((-13682443.000 5091132.739, -13682445...  
1  POLYGON ((-13576727.142 4841226.161, -13576726...  
2  POLYGON ((-13459243.000 4621236.000, -13458968...  
3  POLYGON ((-13468077.000 4642260.000, -13467975...  
4  POLYGON ((-13468418.000 4614853.000, -13468428...

In [3]:
# View information about the perimeter data frame
perimeters.info()

<class 'geopandas.geodataframe.GeoDataFrame'>
RangeIndex: 22261 entries, 0 to 22260
Data columns (total 22 columns):
 #   Column      Non-Null Count  Dtype   
---  ------      --------------  -----   
 0   OBJECTID    22261 non-null  int64   
 1   YEAR_       22184 non-null  float64 
 2   STATE       22261 non-null  object  
 3   AGENCY      22208 non-null  object  
 4   UNIT_ID     22194 non-null  object  
 5   FIRE_NAME   15672 non-null  object  
 6   INC_NUM     21286 non-null  object  
 7   ALARM_DATE  16865 non-null  object  
 8   CONT_DATE   9632 non-null   object  
 9   CAUSE       22237 non-null  float64 
 10  C_METHOD    10140 non-null  float64 
 11  OBJECTIVE   22014 non-null  float64 
 12  GIS_ACRES   22261 non-null  float64 
 13  COMMENTS    2707 non-null   object  
 14  COMPLEX_NA  596 non-null    object  
 15  IRWINID     2695 non-null   object  
 16  FIRE_NUM    17147 non-null  object  
 17  COMPLEX_ID  360 non-null    object  
 18  DECADES     22184 non-null  float64 
 

In [4]:
# Check the perimeter CRS 
perimeters.crs

<Projected CRS: EPSG:3857>
Name: WGS 84 / Pseudo-Mercator
Axis Info [cartesian]:
- X[east]: Easting (metre)
- Y[north]: Northing (metre)
Area of Use:
- name: World between 85.06°S and 85.06°N.
- bounds: (-180.0, -85.06, 180.0, 85.06)
Coordinate Operation:
- name: Popular Visualisation Pseudo-Mercator
- method: Popular Visualisation Pseudo Mercator
Datum: World Geodetic System 1984 ensemble
- Ellipsoid: WGS 84
- Prime Meridian: Greenwich

In [5]:
# Find the min and max year of the df
print(perimeters.YEAR_.min())
print(perimeters.YEAR_.max())

1878.0
2023.0


This fire perimeter data comes from CalFire and includes data for all fire perimeters from 1878 to 2023. It has data on the year, the fire name, the reporting agency, the cause, duration, among other data. The CRS is NAD83 California Albers and it is a projected CRS (EPSG:3310)


Save only the 2017 Thomas Fire boundary as a geospatial file in the format of your choosing. The file should go into the data/ directory in your repository.


In [6]:
# Clean all column names
perimeters.columns = perimeters.columns.str.lower()

# Clean year column name
perimeters = perimeters.rename(columns={'year_': 'year'})

In [7]:
# Select 2017 thomas fire boundary
thomas_boundary = perimeters[(perimeters['year'] == 2017.0) & (perimeters['fire_name'] == 'THOMAS')]
thomas_boundary

objectid    year state agency unit_id fire_name   inc_num  alarm_date  \
2654      2655  2017.0    CA    USF     VNC    THOMAS  00003583  2017-12-04   

       cont_date  cause  c_method  objective  gis_acres  \
2654  2018-01-12    9.0       7.0        1.0   281790.9   

                        comments complex_na irwinid fire_num complex_id  \
2654  CONT_DATE based on Inciweb       None    None     None       None   

      decades    shape__are     shape__len  \
2654   2010.0  1.681106e+09  540531.887458   

                                               geometry  
2654  MULTIPOLYGON (((-13316089.016 4088553.040, -13...

In [8]:
# Export thomas fire boundary as a shape file
# thomas_boundary.to_file(filename='data/thomas_boundary/thomas_boundary.shp', driver='ESRI Shapefile')

thomas_boundary.to_file(filename='data/thomas_boundary.geojson', driver='GeoJSON')


In a markdown cell, briefly explain your reasoning for selecting that specific file format.

I chose to export as a shape file because this is the file format we have used the most in both our geospatial class and in our environmental data class. 